app/components/ModalForm.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { View, Text, TextInput, Modal, TouchableOpacity, Platform, StyleSheet } from 'react-native';
+
+import Button from './Button';
+import colors from '../styles/colors';
+import fonts from '../styles/fonts';
+
+function ModalForm({ animationType = 'fade', visible, title, textInputProps, submitText, onSubmit, onCancel }) {
+  // Our application is not expecting various types of layouts in the forms provided,
+  // thus this component only allows one (or none) text input and one (or none) drop down.
+  // If you want to allow more flexibility in what form fields to allow, simply make use
+  // of the 'children' prop and allow users of this component to add their own fields.
+  // TODO: Implement drop down
+  return (
+    <Modal
+      animationType={animationType}
+      transparent={true}
+      visible={visible}
+    >
+      <View style={styles.modal}>
+        <View style={styles.form}>
+          <Text style={styles.title}>{title}</Text>
+          {textInputProps && (
+            <View style={styles.inputContainer}>
+              <TextInput
+                {...textInputProps}
+                style={styles.input}
+              />
+            </View>
+          )}
+          <Button
+            text={submitText}
+            onPress={onSubmit}
+            otherStyles={{ padding: 10 }}
+          />
+          <TouchableOpacity
+            onPress={onCancel}
+            style={styles.cancelButton}
+          >
+            <Text style={styles.cancelButtonText}>Cancel</Text>
+          </TouchableOpacity>
+        </View>
+      </View>
+    </Modal>
+  );
+}
+
+const styles = StyleSheet.create({
+  modal: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: 'rgba(0,0,0,0.2)'
+  },
+  form: {
+    margin: 30,
```

Comment:
What's the unit in these types of padding/margin values?
If they're pixels, they may not render consistently on devices with different resolutions/screen sizes.

In HTML stylesheets, for example, you would usually use `%` or `em` as units.  I don't know if that applies to React Native as well, though.

Reply:
I'll add to this.  The units aren't exactly pixels.  They do however try to remain consistent across devices.

> There is no universal mapping from points to physical units of measurement. This means that a component with fixed dimensions might not have the same physical size, across different devices and screen sizes. However, this difference is unnoticable for most use cases.